lib/ace/edit_session.js

```diff
@@ -610,7 +610,7 @@ var EditSession = function(text, mode) {
     *
     * @return {Number} The new marker id
     **/
-    this.addMarker = function(range, clazz, type, inFront) {
+    this.addMarker = function(range, clazz, type, inFront, content) {
```

Comment:
Do not use addMarker to add each marker individually, but use addDynamicMarker instead, to add one marker similar to https://github.com/ajaxorg/ace/blob/master/lib/ace/search_highlight.js https://github.com/ajaxorg/ace/blob/master/lib/ace/edit_session.js#L693-L694